### DISCUSSION TITLE
StateNotifierProvider with a nullable notifier

### DISCUSSION THREAD

**Discussion by derolf** [rrousselGit/riverpod#2497] [Q&A]
Consider the following provider:

```
final speechStateProvider =  StateNotifierProvider.autoDispose<SpeechController?, bool?>(...);
```

If "speaking" is not available on the device, I'd like the provider to have a `null` notifier and `null` state.

Of course this currently does not work, do you have a better proposal?

**Comment by rrousselGit** [comment]
This isn't really doable. I'd suggest only having the state as nullable.
